# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/TaskQueueWrapper.h
# Commit: c7dbdd82e8e8
# Full Hash: c7dbdd82e8e8f9cf93fe92dca922755320f36b29
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 3
# Description:
#   Bug 1654112 - Set main thread as current TaskQueue when calling into libwebrtc so that audio calling works. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102286
# ==============================================================================

diff -r 7e00d78d6389 -r c7dbdd82e8e8 dom/media/webrtc/libwebrtcglue/TaskQueueWrapper.h
--- a/dom/media/webrtc/libwebrtcglue/TaskQueueWrapper.h	Tue Feb 16 17:46:07 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/TaskQueueWrapper.h	Mon Feb 01 09:28:23 2021 +0100
@@ -25,6 +25,20 @@
  */
 class TaskQueueWrapper : public webrtc::TaskQueueBase {
  public:
+  class MainAsCurrent {
+#ifdef RELEASE_OR_BETA
+#  error "We must not ship with this class. Main is not a worker thread."
+#endif
+   public:
+    MainAsCurrent() : mSetter(GetMainWorker()) {}
+    ~MainAsCurrent() = default;
+
+   private:
+    CurrentTaskQueueSetter mSetter;
+  };
+
+  static TaskQueueWrapper* GetMainWorker();
+
   explicit TaskQueueWrapper(RefPtr<TaskQueue> aTaskQueue)
       : mTaskQueue(std::move(aTaskQueue)) {}
   ~TaskQueueWrapper() = default;